src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx

```diff
@@ -55,12 +48,18 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
   };
 
   fetchIstioObjectDetailsFromProps = (props: IstioConfigId) => {
-    const promiseConfigDetails = API.getIstioConfigDetail(
-      authentication(),
-      props.namespace,
-      props.objectType,
-      props.object
-    );
+    const promiseConfigDetails = props.objectSubtype
+      ? API.getIstioConfigDetailSubtype(
+          authentication(),
+          props.namespace,
+          props.objectType,
+          props.objectSubtype,
+          props.object
+        )
+      : API.getIstioConfigDetail(authentication(), props.namespace, props.objectType, props.object);
```

Comment:
There is a huge code complexity reduction in that component, that is great. 
However, the idea of handling the behavior differences with `if-else` makes me think that we should explore the idea of composing components. e.g. having one single page that renders a TypeDetails or SubtypeDetails components depending on the base URL. And both components renders one single component that has as props all the basic information to render the details page layout.
Still not really familiar with the react-way though.

Reply:
I'm not sure if we can avoid of the if-else part to extract the object, because in any case, we are dealing with different types and I see only two approaches (with potential variations): to miss the types (as it happens) and to render just a pure yaml, or to create a render component per subtype, which probably it is not worth yet, as we only need the yaml representation.

I think I can refactor a little bit to make it more clear, but I wouldn't enter in major changes as in the next JIRA, plan is to refactor this to enable edition.

Reply:
Don't believe in all you read :-) specially related code style.